src/resources/js/dropdowns.js

```diff
@@ -14,307 +14,361 @@ var tribe_dropdowns = tribe_dropdowns || {};
+				'conditionIsNotNumeric',
+				'condition-is-not-numeric',
+				'conditionChecked',
+				'condition-is-checked'
+			],
+			$container;
+
+		// For Reference we save the jQuery element as an Arg
+		args.$select = $select;
+
+		// Auto define the Width of the Select2
+		args.dropdownAutoWidth = true;
+
+		// CSS for the dropdown
+		args.dropdownCss = {
+			'width': 'auto'
+		};
 
-			if ( 'tribe-ea-field-origin' === $select.attr( 'id' ) ) {
-				args.formatResult = args.upsellFormatter;
-				args.formatSelection = args.upsellFormatter;
-				args.escapeMarkup = function(m) { return m; };
-			}
+		// How do we match the Search
+		args.matcher = obj.matcher;
 
-			if ( $select.is( '[multiple]' ) ) {
-				args.multiple = true;
+		if ( ! $select.is( 'select' ) ) {
+			// Better Method for finding the ID
+			args.id = obj.search_id;
+		}
 
-				if ( ! _.isArray( $select.data( 'separator' ) ) ) {
-					args.tokenSeparators = [ $select.data( 'separator' ) ];
-				} else {
-					args.tokenSeparators = $select.data( 'separator' );
-				}
-				args.separator = $select.data( 'separator' );
-
-				// Define the regular Exp based on
-				args.regexSeparatorElements = [ '^(' ];
-				args.regexSplitElements = [ '(?:' ];
-				$.each( args.tokenSeparators, function ( i, token ) {
-					args.regexSeparatorElements.push( '[^' + token + ']+' );
-					args.regexSplitElements.push( '[' + token + ']' );
-				} );
-				args.regexSeparatorElements.push( ')$' );
-				args.regexSplitElements.push( ')' );
-
-				args.regexSeparatorString = args.regexSeparatorElements.join( '' );
-				args.regexSplitString = args.regexSplitElements.join( '' );
-
-				args.regexToken = new RegExp( args.regexSeparatorString, 'ig' );
-				args.regexSplit = new RegExp( args.regexSplitString, 'ig' );
+		// By default we allow The field to be cleared
+		args.allowClear = true;
+		if ( $select.is( '[data-prevent-clear]' ) ) {
+			args.allowClear = false;
+		}
+
+		// If we are dealing with a Input Hidden we need to set the Data for it to work
+		if ( $select.is( '[data-options]' ) ) {
+			args.data = $select.data( 'options' );
+
+			if ( ! $select.is( 'select' ) ) {
+				args.initSelection = obj.init_selection;
 			}
+		}
 
-			/**
-			 * Better way of matching results
-			 *
-			 * @param  {string} term Which term we are searching for
-			 * @param  {string} text Search here
-			 * @return {boolean}
-			 */
-			args.matcher = function( term, text ) {
-				var result = text.toUpperCase().indexOf( term.toUpperCase() ) == 0;
-
-				if ( ! result && 'undefined' !== typeof args.tags ){
-					var possible = _.where( args.tags, { text: text } );
-					if ( args.tags.length > 0  && _.isObject( possible ) ){
-						var test_value = obj.search_id( possible[0] );
-						result = test_value.toUpperCase().indexOf( term.toUpperCase() ) == 0;
-					}
-				}
+		// Don't Remove HTML elements or escape elements
+		if ( $select.is( '[data-allow-html]' ) ) {
+			args.escapeMarkup = obj.allow_html_markup;
```

Comment:
Here and a few other places it looks like we're assigning the function itself (to `args.escapeMarkup` in this case) rather than the function result - is that deliberate or something that needs to be fixed?